src/trading/CancelOrder.sol

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (C) 2015 Forecast Foundation OU, full GPL notice in LICENSE
+ */
+
+pragma solidity ^0.4.13;
+
+import 'ROOT/Controller.sol';
+import 'ROOT/libraries/ReentrancyGuard.sol';
+import 'ROOT/libraries/Trading.sol';
+import 'ROOT/trading/Cash.sol';
+import 'ROOT/trading/Orders.sol';
+
+
+/**
+ * @title CancelOrder
+ * @dev This allows you to cancel orders on the book.
+ */
+contract CancelOrder is Controlled, ReentrancyGuard {
+    /**
+     * @dev Cancellation: cancels an order, if a bid refunds money, if an ask returns shares
+     * @return true if successful; throw on failure
+     */
+    function cancelOrder(bytes32 _orderId, Trading.TradeTypes _type, Market _market, uint8 _outcome) nonReentrant external returns (bool) {
+        require(_orderId != bytes32(0));
+        require(_market.getTypeName() == "Market");
+
+        // Look up the order the sender wants to cancel
+        Orders _orders = Orders(controller.lookup("Orders"));
+        uint256 _fxpMoneyEscrowed = _orders.getOrderMoneyEscrowed(_orderId, _type, _market, _outcome);
+        uint256 _fxpSharesEscrowed = _orders.getOrderSharesEscrowed(_orderId, _type, _market, _outcome);
+
+        // Check that the order ID is correct, the type is bid (1) or ask (2), and that the sender owns the order
+        require(_type == Trading.TradeTypes.Bid || _type == Trading.TradeTypes.Ask);
```

Comment:
We can probably delete this now that we have an enum.  This was originally intended to protect us in Serpent when this function just took in a number for `_type`.